jack-core/src/com/rapleaf/jack/transaction/TransactorImpl.java

```diff
@@ -96,7 +96,14 @@ DbMetrics getDbMetrics() {
     } catch (Exception e) {
       LOG.error("SQL execution failure", e);
       if (asTransaction) {
-        connection.rollback();
+        try {
+          connection.rollback();
+        } catch (Exception e2) {
+          // ignore. We can't actually do anything about this, and we need to return the connection
```

Comment:
Is it worth logging that we were unable to rollback?

Reply:
Oh, yeah that might be useful. I'll log it.